pkg/utils/utils.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"crypto/sha512"
+	"fmt"
+)
+
+const (
+	maxChainLength = 29
+	chainPrefix    = "CNI-"
+	prefixLength   = len(chainPrefix)
+)
+
+// Generates a chain name to be used with iptables.
+// Ensures that the generated chain name is less than
+// maxChainLength chars in length
+func FormatChainName(name string, id string) string {
+	chainBytes := sha512.Sum512([]byte(name + id))
```

Comment:
It might not be a problem in this specific case, but usually I think it is better to add a separator between fields being hashed to ensure collisions cannot be forged. If "_" is a character that is not allowed in name or id:
```
chainBytes := sha512.Sum512([]byte(name + "_" + id))
```